ServiceWebsite/ServiceWebsite.Services/HearingsService.cs

```diff
@@ -1,28 +1,53 @@
 ﻿using System;
-using System.Collections.Generic;
 using System.Linq;
+using System.Net;
 using System.Threading.Tasks;
+using ServiceWebsite.BookingsAPI.Client;
 using ServiceWebsite.Common;
 using ServiceWebsite.Domain;
 
 namespace ServiceWebsite.Services
 {
     public class HearingsService : IHearingsService
     {
-        public async Task<Hearing[]> GetHearingsFor(string userEmail)
+        private readonly IBookingsApiClient _bookingsApiClient;
+
+        public HearingsService(IBookingsApiClient bookingsApiClient)
         {
-           
-            List<Hearing> hearings = new List<Hearing>();
-            for(int i=1;i<=5;i++)
+            _bookingsApiClient = bookingsApiClient;
+        }
+
+        public async Task<Hearing> GetHearingFor(string username, Guid id)
+        {
+            try
+            {
+                var hearingResponse = await _bookingsApiClient.GetHearingDetailsByIdAsync(id);
+                if (!HearingContainsParticipant(hearingResponse, username))
+                {
+                    throw new UnauthorizedAccessException("User is not participant of hearing: " + id);
+                }
+                return Map(hearingResponse);
+            }
+            catch (BookingsApiException e)
             {
+                if (e.StatusCode == (int) HttpStatusCode.NotFound)
+                {
+                    throw new NotFoundException("Could not find hearing with id: " + id);
```

Comment:
Do we need to throw this, as we would know its a not found from the BookingsApiException ??

Reply:
We shouldn't let the details of the data source bleed out to the controller, that would make the controller dependent on the bookings API, that's why I rethrow with a new type.

Reply:
ok